src/foam/nanos/notification/broadcast/BroadcastNotificationFacade.js

```diff
@@ -29,36 +35,103 @@ foam.CLASS({
   properties: [
     {
       __copyFrom__: 'foam.nanos.notification.Notification.GROUP_ID',
-      label: 'Send To',
-      validateObj: function(groupId) {
-        if ( ! groupId ) {
-          return this.GROUP_REQUIRED;
+      label: 'Send To Group',
+      validateObj: function(groupId, users) {
+        if ( ! groupId && ! ( users && users.length ) ) {
+          return this.GROUP_OR_USERS_REQUIRED;
         }
       },
       view: function(_, X) {
-        const e = foam.mlang.Expressions.create();
-        const group = foam.nanos.auth.Group;
         var dao = X[X.data.GROUP_ID.targetDAOKey] || X.data[X.data.GROUP_ID.name + '$dao'];
         // TODO: find a better way to only pick children
-        let dao$ = foam.core.PromiseSlot.create({
-          promise: X.auth.check(null, '*').then(value => {
-            if ( value ) return dao;
-            return dao.where(e.CONTAINS(group.ID, X.subject.user.spid));
-          })
-        }, X);
-        return { class: 'foam.u2.view.ReferenceView', dao$: dao$ };
+        // dao = dao.where(X.data.CONTAINS(group.ID, X.subject.user.spid));
+        return { class: 'foam.u2.view.ReferenceView', dao: dao };
+      }
+    },
+    {
+      class: 'Array',
+      name: 'users',
+      label: 'Send to Users',
+      view: function(_, X) {
+        var userDAOSlot = X.data.slot(groupId => {
+          if ( groupId ) {
+            return X.userDAO.where(X.data.EQ(X.data.User.GROUP, groupId));
+          } else {
+            return X.userDAO.where(X.data.EQ(X.data.User.SPID, X.subject.user.spid));
+          }
+        });
+        return {
+          class: 'foam.u2.view.ReferenceArrayView',
+          daoKey: 'userDAO',
+          allowDuplicates: false,
+          valueView: {
+            class: 'foam.u2.view.RichChoiceReferenceView',
+            search: true,
+            sections: [
+              {
+                heading: 'Users',
+                dao$: userDAOSlot
+              }
+            ]
+          }
+        }
+      },
+      validateObj: function(groupId, users) {
+        if ( ! groupId && ! ( users && users.length ) ) {
```

Comment:
What if users and group are selected, do we need to notify the user that its incorrect

Reply:
They can both be selected. If the group is selected, it will filter the user list. 
And remember, the audience is an operator.